src/main/java/com/github/onsdigital/dp/uploadservice/api/APIClient.java

```diff
@@ -1,4 +1,132 @@
+
+public class APIClient implements Client  {
+
+    private static final int CHUNK_SIZE = 1024 * 1024 * 5; // 5MB chunks
+
+    private String hostname;
+    private String authToken;
+    private CloseableHttpClient httpClient;
+
+    public APIClient(String hostname, String authToken) {
+        this.hostname = hostname;
+        this.authToken = authToken;
+        httpClient = HttpClients.createDefault();
+    }
+
+    @Override
+    public void uploadFile(File file, List<NameValuePair> params) {
+        URIBuilder uriBuilder;
+        try {
+            uriBuilder = new URIBuilder(hostname);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+        uriBuilder.addParameters(params);
+
+        try {
+            HttpPost request = new HttpPost(uriBuilder.build());
+            final MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+            builder.addBinaryBody("file", file);
+            builder.setBoundary("--TFJ5T8Nl2Py-S_BZXD5_FaEzCCuRXVXL0--[\\r][\\n]");
+            final HttpEntity entityReq = builder.build();
+            request.setEntity(entityReq);
+            request.addHeader("Authorization", "Bearer " + authToken);
+            httpClient.execute(request);
+        } catch (Exception e) {
+            throw new ConnectionException("error talking to upload service", e);
+        }
+    }
+
+    @Override
+    public void uploadResumableFile(File file, List<NameValuePair> params) {
+        long fileSize = file.length();
+        int totalChunks = (int) Math.ceil((double) fileSize / CHUNK_SIZE);
+
+        try (FileInputStream fileInputStream = new FileInputStream(file)) {
+
+            params.add(new BasicNameValuePair("resumableTotalSize", String.valueOf(fileSize))); // the total size of the file
+            params.add(new BasicNameValuePair("resumableTotalChunks", String.valueOf(totalChunks)));
+
+            for (int chunkNumber = 0; chunkNumber < totalChunks; chunkNumber++) {
+                int start = chunkNumber * CHUNK_SIZE;
+                int resumableChunkSize = Math.min(CHUNK_SIZE, (int) (fileSize - start));
+                byte[] buffer = new byte[resumableChunkSize];
+
+                int bytesRead = fileInputStream.read(buffer, 0, resumableChunkSize);
+                if (bytesRead == -1) break;
+
+                params.add(new BasicNameValuePair("resumableChunkNumber", String.valueOf(chunkNumber+1)));
+                params.add(new BasicNameValuePair("resumableChunkSize", String.valueOf(resumableChunkSize)));
+
+                sendChunk(buffer, chunkNumber, totalChunks, file.getName(), params);
+
+                params.removeIf(nameValuePair -> nameValuePair.getName().equals("resumableChunkNumber"));
+                params.removeIf(nameValuePair -> nameValuePair.getName().equals("resumableChunkSize"));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
This should be thrown as an IO exception rather than just printing out the stack trace

Reply:
This is now addressed.